public/js/item-sidebar.js

```diff
@@ -3,12 +3,23 @@ const PST_AMOUNT = document.getElementById('current_pst').value;
 const GST_AMOUNT = document.getElementById('current_gst').value;
 const PURCHASER_ID = document.getElementById('purchaser_id').value;
 
+const APPLY_GIFT_CARD_BUTTON = document.getElementById('apply_gift_card');
+const GIFT_CARD_CODE_INPUT = document.getElementById('gift_card_code_input');
+let GIFT_CARD = null;
+
 window.onload = () => {
     const storedItems = localStorage.getItem(`items-${PURCHASER_ID}`);
     if (storedItems !== null) {
         ITEMS = JSON.parse(storedItems);
     }
 
+    const storedGiftCard = localStorage.getItem(`gift-card-${PURCHASER_ID}`);
+    if (storedGiftCard !== null) {
+        GIFT_CARD = JSON.parse(storedGiftCard);
```

Comment:
Make another call back to the api to make sure it's balance hasn't changed + update the table row with fresh balance